pkg/discovery/dtofactory/node_group_entity_dto_builder.go

```diff
@@ -29,17 +30,29 @@ func NewNodeGroupEntityDTOBuilder(clusterSummary *repository.ClusterSummary, oth
 // Build entityDTOs based on the given node list.
 func (builder *nodeGroupEntityDTOBuilder) BuildEntityDTOs() ([]*proto.EntityDTO, error) {
 	var result []*proto.EntityDTO
-	nodeGrp2nodes := make(map[string]sets.String) // Map of nodeGroup ---> nodes
+	nodeGrp2nodes := make(map[string]sets.String)     // Map of nodeGroup ---> nodes
+	nodeGrp2workloads := make(map[string]sets.String) // Map of nodeGroup ---> workloads of all pods on each node of the nodegroup
 	for _, node := range builder.clusterSummary.Nodes {
+		allPods := append(builder.clusterSummary.GetPendingPodsOnNode(node), builder.clusterSummary.GetRunningPodsOnNode(node)...)
 		for key, value := range node.ObjectMeta.Labels {
 			fullLabel := key + "=" + value
-			nodeLst := nodeGrp2nodes[fullLabel]
-			if nodeLst.Len() == 0 {
-				nodeLst = sets.NewString(string(node.UID))
-			} else {
-				nodeLst.Insert(string(node.UID))
+
+			if _, exists := nodeGrp2nodes[fullLabel]; !exists {
+				nodeGrp2nodes[fullLabel] = sets.NewString()
+			}
+			nodeGrp2nodes[fullLabel].Insert(string(node.UID))
+
+			//add code for nodeGrp2workload
```

Comment:
All looks good to me except for this comment. Maybe simply remove it.

Reply:
Removed. Thanks for the help for reviewing this.